### ISSUE TITLE
Extra newline(s) with `:!` (shell)

### ISSUE THREAD

**Issue by blueyed** [neovim/neovim#7788]
`:!echo 1` displays an extra newline in the beginning, but not the one at the end

```
:!echo 1

1
Press ENTER or type command to continue
```

With Vim it looks like this:
```
1

Press ENTER or type command to continue
```

Using `:!printf 1` (which produces no newline at the end) results in the same (unexpected) output with Neovim, but the expected with Vim:
```
1
Press ENTER or type command to continue
```

- `nvim --version`: v0.2.3-330-g5dd2ca767
- Vim (version: ) behaves differently? Yes

Tried it with both `&shell` being `zsh` and `bash`.

I'm quite sure this is a known issue, but could not find an issue that tracks it.

**Comment by justinmk** [comment]
It's been like this forever, I am certain there is no technical reason for it, rather we just didn't think it mattered very much. Could modify `out_data_append_to_screen()` to better match Vim's behavior.

**Comment by nimitbhardwaj** [comment]
Hey, perhaps this issue is fixed, my `nvim --version`: `NVIM v0.2.3-703-g0f1bc5ddc`, giving expected output
